custom_components/ha-google-home/api.py

```diff
@@ -104,10 +116,26 @@ def get_google_devices_information(self):
                             device=device.device_name, error=API_RETURNED_UNKNOWN
                         )
                     )
+            else:
+                offline_devices.append(device.device_name)
 
             device.timers = timers
             device.alarms = alarms
             _LOGGER.debug(device)
 
+        """Gives the user a warning if the device is offline, but will not remove entities or device from
+        HA device registry"""
+        if len(offline_devices) > 0:
+            if_one = (
+                "A device is " if len(offline_devices) == 1 else "Some devices are "
+            )
+            _LOGGER.warning(
+                "{one_or_more}offline, so no information could be retrieved, will try again later - {devices}".format(
+                    one_or_more=if_one,
+                    devices=self.format_offline_devices_to_human_string(
+                        offline_devices
+                    ),
+                )
+            )
```

Comment:
Instead of `offline_devices` list conversion to `str`, which is a bit complex, I propose we do a`for` loop and print a new warning per each device, just informing that the device is offline. I think it will take down a lot of unnecessary complexity away :)

Reply:
The reason I did it this way was because that sometime the request to the device will fail for no good reason, so to not overflow the user with warnings, i thought printing it on one line instead. I'm also working on a way to check if an IP is valid or not.

Reply:
After thinking about it, your right, i have changed :smile: